build.gradle

```diff
@@ -354,6 +354,8 @@ subprojects { subproj ->
     }
 
     test {
+        maxParallelForks = Runtime.runtime.availableProcessors() ?: 1
```

Comment:
huh, I could have sworn this was already set from a while back

Reply:
from the documentation, maybe  `maxParallelForks = Runtime.runtime.availableProcessors().intdiv(2) ?: 1` is better?

otherwise, if run locally, might suck up alot of a machine's resources.

either way easy to change, so not doing the divide is also good!

Reply:
are we sure this will actually make a difference? 

@git-phu and i played with this late last year and our essential takeaway was that gradle was already basically parallelizing as best as it could overall in builds (parallel behavior is set at the top level build.gradle alread)

Reply:
OH this is interesting. So apparently there are two levels of parallelism here with gradle. One is at the task level, and one is at the test level. The one sergio is modifying is at the test level, and the one @git-phu and I played with is the task level parallelism: https://discuss.gradle.org/t/relationship-between-forkevery-maxparallelfork-and-parallel/25126

Reply:
yeah! the [smallest unit of parallelism is still the test class](https://kousenit.org/2020/11/22/simple-demo-of-gradle-parallel-tests/) so for the platform build we'll probably need to do some work splitting up the bigger test files that exists.

actually @sergio-ropero probably a good idea to add a comment explaining this since I'd imagine other devs will have the same questions as conor

Reply:
one other somewhat interesting implication of this is that peter and i played with the max-workers number and it didn't seem to affect much back then. But I wonder whether playing with max-workers with forks in a test would actually make a difference after sergio's changes when targeting these specific builds.

Reply:
> from the documentation, maybe maxParallelForks = Runtime.runtime.availableProcessors().intdiv(2) ?: 1 is better?
> otherwise, if run locally, might suck up alot of a machine's resources.

To be honest I thought the same... probably we can add a logic if it's local execution or remote. For me, makes sense to use as much as possible of the remote instances... they just span to execute the test, so using half of them is not good.

I have tested in local and I have not feel like getting out of resources.
Also, to be hones, even if there were nodes available, not all of them were used. I assume that at some point, only a few classes runs in parallel.
Also, tests takes around 1 minute... so I think is fine

> actually @sergio-ropero probably a good idea to add a comment explaining this since I'd imagine other devs will have the same questions as conor

Yeah... At the beginning I was thinking about parallelise both test (unit test and integration test... even acceptance test) but as I saw in the gradle configuration files, there is an explicit dependency... so I assumed that running unit before integration was intended. Also, I think that makes sense, as usually integration tests are more time consuming, and if unit tests are wrong, there is no reason to run integration tests. Open to discuss it with you all 😊